experiments/snippets/skuidCustom/skuidCustom.js

```diff
@@ -0,0 +1,1344 @@
+								let enclose = false;
+								let ourValue = value;
+
+								if (displayTypesToEnclose.includes(model.fieldsMap[key].displaytype)) {
+									enclose = true;
+								}
+
+								if (ourValue === null || ourValue === 'null') {
+									if (enclose) {
+										ourValue = '';
+									} else {
+										ourValue = 'NULL';
+									}
+								}
+
+								if (!first) {
+									thisCondition += ' AND ';
+								}
+
+								if (enclose) {
+									thisCondition += `${key} = '${ourValue}'`;
+								} else {
+									thisCondition += `${key} = ${ourValue}`;
+								}
+							}
+						}
+
+						first = false;
+					});
+
+					// add to our lastRowConditions
+					if (thisCondition !== '' && !dontQueryNextBoolean) {
+						if (lastRowConditions === '') {
+							lastRowConditions = '(';
+						}
+						lastRowConditions += `${orCondition}(${thisCondition})`;
+					}
+
+					// pop the last field to condition on
+					lrcArr.pop();
+				}
+
+				if (lastRowConditions !== '') {
+					lastRowConditions += ')';
+				}
+			}
+		}
+
+		let lastRowConditionsPre = '';
+
+		// If we have Last Row conditions, need to add AND
+		if (lastRowConditions !== '' && ourConditionStr !== '') {
+			lastRowConditionsPre = ' AND ';
+		}
+
+		// If we have conditions, need to add WHERE
+		if (ourConditionStr !== '' || lastRowConditions !== '') {
+			ourConditionStrPre = ' WHERE ';
+		}
+
+		let groupbyPre = '';
+
+		if (groupby !== '') {
+			groupbyPre = ' GROUP BY ';
+		}
+
+		// Construct our HAVING strings
+		let havingPre = '';
+
+		if (model.havings !== undefined && model.havings.length > 0) {
+			for (let i = 0; i < model.havings.length; i++) {
+				const h = model.havings[i];
+				const encloseValueInQuotes = h.encloseValueInQuotes;
+				const fieldfunction = h.fieldfunction || '';
+				const field = h.field;
+				const value = h.value;
+				const operator = operatorConvert(h.operator);
+				let fieldPre = '';
+				let fieldPost = '';
+				let valuePre = '';
+				let valuePost = '';
+
+				// skip inactive fields
+				if (h.inactive === true) {
+					continue;
+				} else {
+					havingPre = ' HAVING ';
+				}
+
+				if (having === '') {
+					having = '(';
+				} else {
+					having += ' AND ';
+				}
+
+				if (fieldfunction !== '') {
+					fieldPre = '(';
+					fieldPost = ')';
+				}
+
+				if (encloseValueInQuotes === true) {
+					valuePre = '\'';
+					valuePost = '\'';
+				}
+
+				having += `(${fieldfunction}${fieldPre}${field}${fieldPost} ${operator} ${valuePre}${value}${valuePost})`;
+			}
+
+			if (having !== '') {
+				having += ')';
+			}
+		}
+
+		const queryStr = `SELECT ${fieldsStr} FROM ${model.objectName}${ourConditionStrPre}${ourConditionStr}${lastRowConditionsPre}${lastRowConditions}${groupbyPre}${groupby}${havingPre}${having}${orderby} LIMIT ${limit}`;
+
+		// console.log('QUERY:');
+		// console.log(queryStr);
+
+		skuid.$.when(skuid.sfdc.api.query(queryStr)).done(function (queryResult) {

```

Comment:
minor, you don't need to wrap this in `skuid.$.when()`. 
```suggestion
		skuid.sfdc.api.query(queryStr).done(function (queryResult) {

```

Reply:
removed the skuid.$.when